tokenbucket.go

```diff
@@ -465,3 +469,132 @@ func (t *TokenBucketRedis) SetState(ctx context.Context, state TokenBucketState)
 
 	return errors.Wrap(err, "failed to save keys to redis")
 }
+
+// TokenBucketDynamoDB is a DynamoDB implementation of a TokenBucketStateBackend.
+type TokenBucketDynamoDB struct {
+	client        *dynamodb.Client
+	tableProps    DynamoDBTableProperties
+	partitionKey  string
```

Comment:
How is this `partitionKey` different from the `tableProps.PartitionKeyName`?

Reply:
See above comment